### PR TITLE
HPCC-15428 CLocalEnvironment doesn't check properly the drop zone path in  getDropZoneByAddressPath() function.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add path separator character at the end of the drop zone path (if there
is not any) before compare it to the target file path.

Signed-off-by: Attila Vamos attila.vamos@gmail.com

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please review.

**Comment by jakesmith** [comment]
The users of the interface don't really want to be concerned with knowing about path separators.
So in fact I think it should cope with either..
i.e. getDropZoneByAddressPath(<ip>, /a/b/c/ ) or getDropZoneByAddressPath(<ip>, /a/b/c )
should work.

Should deal with the trailing separator internally.

@AttilaVamos - back to you for now.

**Comment by AttilaVamos** [comment]
@jakesmith reworked, please re-review.

File: common/environment/environment.cpp [hunk 1]
```diff
@@ -1576,12 +1576,13 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
 
     ForEach(*zoneIt)
     {
-        SCMStringBuffer dropZoneName;
-        zoneIt->query().getName(dropZoneName);
-
         SCMStringBuffer dropZoneDir;
         zoneIt->query().getDirectory(dropZoneDir);
-        const char * pdropzoneDir = dropZoneDir.str();
+        StringBuffer fullDropZoneDir(dropZoneDir.str());
```
Comment (richardkchapman) [inline common/environment/environment.cpp]: Should use addPathSepChar function (in file.hpp)

**Comment by richardkchapman** [comment]
@AttilaVamos One comment

File: common/environment/environment.cpp (hunk 1 above)
Comment (richardkchapman) [inline common/environment/environment.cpp]: (assuming this code is doing what I think it is doing...)

File: common/environment/environment.cpp [hunk 2]
```diff
@@ -1576,12 +1576,13 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
...
+        const char pathSeparator = fullDropZoneDir.charAt(0);
+        if (fullDropZoneDir.charAt(fullDropZoneDir.length()) != pathSeparator)
```
Comment (ghalliday) [inline common/environment/environment.cpp]: Surely this should be charAt(fullDropZoneDir.length()-1.  As Richard says though, better to call the function in jlib.

**Comment by AttilaVamos** [comment]
@richardkchapman reworked, please check.

**Pull request merged**